ts/Core/Axis/PlotLineOrBand.ts

```diff
@@ -452,35 +440,61 @@ class PlotLineOrBand {
              */
             plotLine.label = label = renderer
                 .text(
-                    labelText as any,
+                    this.getLabelText(optionsLabel),
                     0,
                     0,
                     optionsLabel.useHTML
                 )
-                .attr(attribs)
+                .attr({
+                    align: optionsLabel.textAlign || optionsLabel.align,
+                    rotation: optionsLabel.rotation,
+                    'class': 'highcharts-plot-' + (isBand ? 'band' : 'line') +
+                        '-label ' + (optionsLabel.className || ''),
+                    zIndex
+                })
                 .add();
 
             if (!this.axis.chart.styledMode) {
-                label.css(optionsLabel.style as any);
+                label.css(merge({
+                    textOverflow: 'ellipsis'
+                }, optionsLabel.style));
             }
         }
 
         // get the bounding box and align the label
         // #3000 changed to better handle choice between plotband or plotline
-        xBounds = (path as any).xBounds ||
+        const xBounds = (path as any).xBounds ||
             [path[0][1], path[1][1], (isBand ? path[2][1] : path[0][1])];
-        yBounds = (path as any).yBounds ||
+        const yBounds = (path as any).yBounds ||
             [path[0][2], path[1][2], (isBand ? path[2][2] : path[0][2])];
 
-        x = arrayMin(xBounds);
-        y = arrayMin(yBounds);
+        const x = arrayMin(xBounds);
+        const y = arrayMin(yBounds);
 
         label.align(optionsLabel, false, {
             x: x,
             y: y,
             width: arrayMax(xBounds) - x,
             height: arrayMax(yBounds) - y
         });
+
+        const rotation = label.rotation || 0;
+        let width = 0;
+
+        if (!rotation) {
+            width = axis.width - (label.alignAttr.x - axis.left);
+        } else if (rotation === 90) {
+            width = axis.height - (label.alignAttr.y - axis.top);
+        } else if (rotation === 180) {
+            width = label.alignAttr.x - axis.left;
+        } else if (rotation === 270) {
+            width = label.alignAttr.y - axis.top;
+        }
+
+        if (width) {
+            label.css({ width: width + 'px' });
+        }
```

Comment:
Since we're already computing the box height and width further up, I think this can be simplified. The following works for me, but it doesn't take into account the rotation of 180 and 270 - but I am not able to make those work in the first place...

```js
        const x = arrayMin(xBounds);
        const y = arrayMin(yBounds);
        const width = arrayMax(xBounds) - x;
        const height = arrayMax(yBounds) - y;

        label.align(optionsLabel, false, {
            x: x,
            y: y,
            width,
            height
        });

        label.css({
            width: (
                axis.horiz ?
                    height - (optionsLabel.y || 0) :
                    width - (optionsLabel.x || 0)
            ) + 'px'
        });
        ```

Reply:
Hmm, heres 180: https://jsfiddle.net/epLx81bq/2/

We should possibly handle right-aligned labels as well?

Reply:
> Hmm, heres 180

I'm not sure it's useful though. It is also possible to set any rotation: https://jsfiddle.net/highcharts/krogm2u4/ . To keep it simple, maybe we should just stick with rotations of 0 and 90?

> We should possibly handle right-aligned labels as well?

Yes, possibly. But I'm afraid of bloat. Just wondering if there is any synergy with other similar concepts here. Chart title and subtitle perhaps.